Python/initconfig.c

```diff
@@ -3261,3 +3388,503 @@ _Py_DumpPathConfig(PyThreadState *tstate)
+    case PyConfig_MEMBER_INT:
+    case PyConfig_MEMBER_UINT:
+    {
+        int value = *(int *)member;
+        return PyLong_FromLong(value);
+    }
+
+    case PyConfig_MEMBER_BOOL:
+    {
+        int value = *(int *)member;
+        return PyBool_FromLong(value != 0);
+    }
+
+    case PyConfig_MEMBER_ULONG:
+    {
+        unsigned long value = *(unsigned long *)member;
+        return PyLong_FromUnsignedLong(value);
+    }
+
+    case PyConfig_MEMBER_WSTR:
+    case PyConfig_MEMBER_WSTR_OPT:
+    {
+        wchar_t *wstr = *(wchar_t **)member;
+        if (wstr != NULL) {
+            return PyUnicode_FromWideChar(wstr, -1);
+        }
+        else {
+            return Py_NewRef(Py_None);
+        }
+    }
+
+    case PyConfig_MEMBER_WSTR_LIST:
+    {
+        if (strcmp(spec->name, "xoptions") == 0) {
+            return _PyConfig_CreateXOptionsDict(config);
+        }
+        else {
+            const PyWideStringList *list = (const PyWideStringList *)member;
+            PyObject *pylist = _PyWideStringList_AsList(list);
+            if (pylist == NULL) {
+                return NULL;
+            }
+
+            PyObject *tuple = PyList_AsTuple(pylist);
+            Py_DECREF(pylist);
+            return tuple;
+        }
+    }
+    default:
+        PyErr_Format(PyExc_TypeError,
+                     "config option %s is not a strings list", spec->name);
+        return NULL;
+    }
+}
+
+
+static PyObject*
+preconfig_get(const PyPreConfig *preconfig, const PyConfigSpec *spec)
+{
+    // The type of all PYPRECONFIG_SPEC members is INT or BOOL.
+    assert(spec->type == PyConfig_MEMBER_INT
+           || spec->type == PyConfig_MEMBER_BOOL);
+
+    char *member = (char *)preconfig + spec->offset;
+    int value = *(int *)member;
+
+    if (spec->type == PyConfig_MEMBER_BOOL) {
+        return PyBool_FromLong(value != 0);
+    }
+    else {
+        return PyLong_FromLong(value);
+    }
+}
+
+
+PyObject*
+PyConfig_Get(const char *name)
+{
+    const PyConfigSpec *spec = config_find_spec(name);
+    if (spec != NULL) {
+        const PyConfig *config = _Py_GetConfig();
+        return config_get(config, spec, 1);
+    }
+    PyErr_Clear();
+
+    spec = preconfig_find_spec(name);
+    if (spec != NULL) {
+        const PyPreConfig *preconfig = &_PyRuntime.preconfig;
+        return preconfig_get(preconfig, spec);
+    }
+
+    return NULL;
+}
+
+
+int
+PyConfig_GetInt(const char *name, int *value)
+{
+    assert(!PyErr_Occurred());
+
+    PyObject *obj = PyConfig_Get(name);
+    if (obj == NULL) {
+        return -1;
+    }
+
+    if (!PyLong_Check(obj)) {
+        Py_DECREF(obj);
+        PyErr_Format(PyExc_TypeError, "config option %s is not an int", name);
+        return -1;
+    }
+
+    int as_int = PyLong_AsInt(obj);
+    Py_DECREF(obj);
+    if (as_int == -1 && PyErr_Occurred()) {
+        PyErr_Format(PyExc_OverflowError,
+                     "config option %s value doesn't fit into int", name);
```

Comment:
```suggestion
                     "config option %s value does not fit into an int", name);
```

(so that the error messages use non-contracted negation in both cases)